### PR TITLE
New Preact Implementation

### PR DESCRIPTION
**Pull request by malbernaz**

### CODE REVIEW

File: site/_apps/preact-hacker-news.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+weight: 16
```
Comment (housseindjirdeh) [inline site/_apps/preact-hacker-news.txt]: Minor nitpick: could you update the weights to show after [the most recent app](https://github.com/tastejs/hacker-news-pwas/commit/c54aaff35cb5adae88b8ad749b8c1d505ddd3289) (this should be the 22nd)

File: site/_contributors/malbernaz.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+weight: 16
```
Comment (housseindjirdeh) [inline site/_contributors/malbernaz.txt]: Similarly here, just need to update the weight to after our latest contributor :)

**Comment by housseindjirdeh** [comment]
One thing I noticed is Lighthouse screenshot traces don't seem to align with `First Meaningful Paint` times and I'm curious if server-rendered content causes a slight discrepancy with LH 🤔 

image

**Comment by addyosmani** [comment]
> One thing I noticed is Lighthouse screenshot traces don't seem to align with First Meaningful Paint times and I'm curious if server-rendered content causes a slight discrepancy with LH

There's definitely the odd occasion when server-rendering + a long pause before JS gets fetched and executed can cause some discrepancies in the scoring. I'll take another look through the traces now just to check if there's anything I can spot in there that's out of place.

**Comment by addyosmani** [comment]
Traces look about right. As @housseindjirdeh mentions SSR hits first meaningful paint quite a bit earlier than the app is interactive, but I think this might be more of a Lighthouse oddity than the app itself.

One thing I noticed was some of the preloaded scripts don't appear to be getting used that early on. I might suggest not preloading the scripts that are cited here:

image

File: site/_apps/preact-hacker-news.txt [hunk 3]
```diff
@@ -18,7 +18,7 @@ interactive-em: '5s'
 interactive-faster-3g: '4s'
 lighthouse-link: https://www.webpagetest.org/lighthouse.php?test=170710_YP_cb200c6802242f8ad8b7f24e7c916887&run=1
 wpt-em-link: https://www.webpagetest.org/result/170710_25_208c0e8d6dd5d2f60d01b97a963634f7/
-wpt-faster-3g-link: https://www.webpagetest.org/result/170710_YP_cb200c6802242f8ad8b7f24e7c916887/
+wpt-faster-3g-link: https://www.webpagetest.org/result/170714_ZD_baecb5715968cd54b7059efe6489eefa/
 image: /assets/images/preact-hnpwa.png
```
Comment (addyosmani) [inline site/_apps/preact-hacker-news.txt]: Minor feedback about preload use later down.

**Comment by malbernaz** [comment]
hey, folks sorry for the delay! Thank you for your the feedback! @addyosmani! I'll follow your advise and update the app not to preload those resources. And @housseindjirdeh I'll update the weights.

**Comment by malbernaz** [comment]
Hey @addyosmani, I've updated the app not to load unused resources earlier. If by chance you have the time and the will I'd love to hear any other advice that could help me to make the first time to interaction faster, I've been trying to figure out what's happening with no success. @housseindjirdeh will run the tests again tomorrow and update the PR. Thank you all!

**Comment by anubhav7495** [comment]
[LightHouse Faster 3G](https://www.webpagetest.org/lighthouse.php?test=170818_9W_cd328ee0370e86c069c7bd0a70db5e34&run=2) TTI 3.4s

[LightHouse 3G_EM](https://www.webpagetest.org/lighthouse.php?test=170818_R4_717b39b3ee0d52ba738f0830742c82b4&run=1) TTI 4.8s

Please update the times in your PR.

Also, I noticed the app is showing 20 stories per page. Ideally, it should be 30 stories per page as per requirements.

Thanks.

**Comment by addyosmani** [comment]
Thanks for re-running the WPTs for this one, @anubhav7495. 👍 from my side once we've got the scores and ideally pagination updated there. Almost good to go.

**Comment by malbernaz** [comment]
@anubhav7495 you have the links to the WPTs? Above are the lighthouse ones. Thanks anyway.

**Comment by malbernaz** [comment]
Updated links and weights, and now the application has 30 items per page.

File: site/_apps/preact-hacker-news.txt (hunk 3 above)
Comment (addyosmani) [inline site/_apps/preact-hacker-news.txt]: Confirming that the new benchmark numbers here LGTM.

File: site/_apps/preact-hacker-news.txt [hunk 4]
```diff
@@ -14,11 +14,11 @@ hosting: 'now.sh'
 authors:
```
Comment (addyosmani) [inline site/_apps/preact-hacker-news.txt]: Out of interest did this implementation start using Preact CLI or was it from scratch? Just asking as it it was the former would be good to capture that. No worries if not however :)

Comment (malbernaz) [inline site/_apps/preact-hacker-news.txt]: No, it was built from scratch : )

**Comment by malbernaz** [comment]
Is there something preventing this from being pushed?

cc @anubhav7495 @addyosmani

File: site/_apps/preact-hacker-news.txt (hunk 4 above)
Comment (addyosmani) [reply site/_apps/preact-hacker-news.txt]: Thanks for the confirm!

**Pull request merged**